.CMake/alg_support.cmake

```diff
@@ -66,6 +66,9 @@ cmake_dependent_option(OQS_USE_SHA2_OPENSSL "" ON "OQS_USE_OPENSSL" OFF)
 # enough to support our incremental API.
 cmake_dependent_option(OQS_USE_SHA3_OPENSSL "" OFF "OQS_USE_OPENSSL" OFF)
```

Comment:
I am no certain about this line: by default `OQS_USE_SHA3_OPENSSL` is `OFF`, **and** if `OQS_USE_OPENSSL=ON` **also** turns it `OFF`?

Reply:
Hmm .. somewhat surprising instruction, indeed. I presume this had been `ON` until the incremental API got added and @jschanck simply toggled the original "ON" when adding the comment above. I'm inclined to let it remain as-is as a reminder that we may want to improve this as OpenSSL's APIs change to make the comment incorrect (i.e., to allow use of OpenSSL for SHA3_inc operations).

However, given that the incremental API _is_ implemented also for SHA3 and OpenSSL, maybe @jschanck could state more explicitly here what he actually meant by the comment (and whether my assumption above is correct)?

Reply:
> > I'm inclined to let it remain as-is  .  .  .
> 
> Leaving it as-is completely prevents use of OpenSSL SHA3 - there'd be no way to enable it. Is that what you want?

That might have been the goal of @jschanck :) but it definitely isn't the case: Passing `-DOQS_USE_SHA3_OPENSSL=ON` to `cmake` enables `liboqs` to use OpenSSLs SHA3:

```
$ ./tests/test_sha3 
Configuration info
==================
Target platform:  x86_64-Linux-5.4.0-146-generic
Compiler:         gcc (7.5.0)
Compile options:  [-Wa,--noexecstack;-O3;-fomit-frame-pointer;-fdata-sections;-ffunction-sections;-Wl,--gc-sections;-Wbad-function-cast]
OQS version:      0.8.0-dev
Git commit:       d704da0c5fd0e6d22d2a2f4212910cf383d8c345
OpenSSL enabled:  Yes (OpenSSL 1.1.1  11 Sep 2018)
AES:              NI
SHA-2:            OpenSSL
SHA-3:            OpenSSL
OQS build flags:  OQS_DIST_BUILD OQS_OPT_TARGET=generic CMAKE_BUILD_TYPE=Release 
CPU exts active:  AES AVX PCLMULQDQ POPCNT SSE SSE2 SSE3
Success! passed sha3-256 known answer tests 
Success! passed sha3-512 known answer tests 
Success! passed shake-128 known answer tests 
Success! passed shake-256 known answer tests 
Success! passed four-way parallel shake-128 known answer tests 
Success! passed four-way parallel shake-256 known answer tests 

```

Reply:
> but it definitely isn't the case: Passing `-DOQS_USE_SHA3_OPENSSL=ON` to `cmake` enables `liboqs` to use OpenSSLs SHA3

Why doesn't that line disable `OQS_USE_SHA3_OPENSSL` when it sees that `OQS_USE_OPENSSL` is set?

Reply:
>  is that this command after all is/declares/initializes an _option_, i.e., a variable that can be ultimately set on the command line

But this value _is_ an option - so, the only question is whether setting a value in the command line overrides what's the `CMakeLists.txt` content says. Your example seems to show that passing via CLI does override, but I still have my doubts.